interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,574 @@
+        } catch (Exception e) {
+          e.printStackTrace();
+          throw new RuntimeException(e);
+        }
+
+        testCasesSinceLastMetricsCollection++;
+
+        if (lastMetricsCollectionTime
+            + SECONDS.toNanos(METRICS_COLLECTION_INTERVAL_SECS) <= System.nanoTime()) {
+          double durationSeconds = (System.nanoTime() - lastMetricsCollectionTime) / 1000000000.0;
+          long qps = (long) Math.ceil(testCasesSinceLastMetricsCollection / durationSeconds);
+
+          Metrics.GaugeResponse gauge = Metrics.GaugeResponse
+              .newBuilder()
+              .setName(gaugeName)
+              .setLongValue(qps)
+              .build();
+
+          gauges.put(gaugeName, gauge);
+
+          lastMetricsCollectionTime = System.nanoTime();
+          testCasesSinceLastMetricsCollection = 0;
+        }
+      }
+    }
+
+    private void runTestCase(Tester tester, TestCases testCase) throws Exception {
+      // TODO(buchgr): Implement tests requiring auth, once C++ supports it.
+      switch (testCase) {
+        case EMPTY_UNARY:
+          tester.emptyUnary();
+          break;
+
+        case LARGE_UNARY:
+          tester.largeUnary();
+          break;
+
+        case CLIENT_STREAMING:
+          tester.clientStreaming();
+          break;
+
+        case SERVER_STREAMING:
+          tester.serverStreaming();
+          break;
+
+        case PING_PONG:
+          tester.pingPong();
+          break;
+
+        case EMPTY_STREAM:
+          tester.emptyStream();
+          break;
+
+        case UNIMPLEMENTED_METHOD: {
+          tester.unimplementedMethod();
+          break;
+        }
+
+        case CANCEL_AFTER_BEGIN: {
+          tester.cancelAfterBegin();
+          break;
+        }
+
+        case CANCEL_AFTER_FIRST_RESPONSE: {
+          tester.cancelAfterFirstResponse();
+          break;
+        }
+
+        case TIMEOUT_ON_SLEEPING_SERVER: {
+          tester.timeoutOnSleepingServer();
+          break;
+        }
+
+        default:
+          throw new IllegalArgumentException("Unknown test case: " + testCase);
+      }
+    }
+
+    class Tester extends AbstractInteropTest {
+      @Override
+      protected ManagedChannel createChannel() {
+        return Worker.this.channel;
+      }
+    }
+
+    class WeightedTestCaseSelector {
+      // Randomly shuffled list that contains each testcase as often as
+      // the testcase's weight.
+      final List<TestCases> testCases = new ArrayList<TestCases>();
+      int idx;
+
+      WeightedTestCaseSelector(List<TestCaseWeightPair> testCaseWeightPairs) {
+        for (TestCaseWeightPair testCaseWeightPair : testCaseWeightPairs) {
+          for (int i = 0; i < testCaseWeightPair.weight; i++) {
+            testCases.add(testCaseWeightPair.testCase);
+          }
+        }
+
+        shuffle(testCases);
+      }
+
+      TestCases nextTestCase() {
+        return testCases.get(idx++ % testCases.size());
```

Comment:
You can use Guava's Iterators.cycle and not have to save the index (or even the array!)